lib/auth/init.go

```diff
@@ -477,6 +485,24 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 	return asrv, nil
 }
 
+// shouldUpdateResource determines whether the candidate resource
+// should be used to replace the currently stored resource.
+//
+// For more details, see Section "Auth server initialization"
+// in rfd/0016-dynamic-configuration.md
+func shouldUpdateResource(candidate services.ResourceWithOrigin, stored services.ResourceWithOrigin, getErr error) (bool, error) {
```

Comment:
Yeah but then the `trace.IsNotFound` would have to be evaluated outside of this function, right? My thinking was to make this function as generic as possible so that other `ResourceWithOrigin` can do without as less code duplication as possible.